pac4j-oauth/src/main/java/org/pac4j/oauth/client/WechatClient.java

```diff
@@ -0,0 +1,91 @@
+package org.pac4j.oauth.client;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.pac4j.oauth.profile.wechat.WechatProfile;
+import org.pac4j.oauth.profile.wechat.WechatProfileCreator;
+import org.pac4j.oauth.profile.wechat.WechatProfileDefinition;
+import org.pac4j.scribe.builder.api.WechatApi20;
+
+/**
+ * <p>This class is the OAuth client to authenticate users in Tencent Wechat.</p>
+ * <p>It returns a {@link WechatProfile}.</p>
+ * <p>More info at: <a href=
+ * "https://open.weixin.qq.com/cgi-bin/showdocument?action=dir_list&t=resource/res_list&verify=1&id=open1419316505&token=&lang=zh_CN">
+ * WeChat login development guide</a></p>
+ *
+ * @author zhangzhenli
+ * @since 3.1.0
+ */
+public class WechatClient extends OAuth20Client<WechatProfile> {
+
+    public enum WechatScope {
+        /**
+         * Only for WeChat QRCode login. Get the nickname, avatar, and gender of the logged in user.
+         */
+        snsapi_login,
```

Comment:
Can we put that in uppercase and use the appropriate the `toLowerCase()` method in the `getOAuthScope` method?